native/acter/src/api/push/notification_item.rs

```diff
@@ -721,55 +258,34 @@ impl NotificationItemInner {
                 }
                 _ => None,
             },
-            NotificationItemInner::Attachment { content, .. } => content.clone(),
-            NotificationItemInner::Comment { content, .. } => Some(MsgContent::from(content)),
-            NotificationItemInner::Boost {
-                first_slide: Some(first_slide),
-                ..
-            } => match &first_slide {
-                // everything else we have to fallback to the body-text thing ...
-                NewsContent::Fallback(FallbackNewsContent::Text(msg_content))
-                | NewsContent::Text(msg_content) => Some(MsgContent::from(msg_content)),
-                NewsContent::Fallback(FallbackNewsContent::Video(msg_content))
-                | NewsContent::Video(msg_content) => Some(MsgContent::from(msg_content)),
-                NewsContent::Fallback(FallbackNewsContent::Audio(msg_content))
-                | NewsContent::Audio(msg_content) => Some(MsgContent::from(msg_content)),
-                NewsContent::Fallback(FallbackNewsContent::File(msg_content))
-                | NewsContent::File(msg_content) => Some(MsgContent::from(msg_content)),
-                NewsContent::Fallback(FallbackNewsContent::Location(msg_content))
-                | NewsContent::Location(msg_content) => Some(MsgContent::from(msg_content)),
+            NotificationItemInner::Activity(activity) => match activity.content() {
```

Comment:
as the core types are properly typed, we have to wrap the inner content into the local ffi-api of `MsgContent` here